service/lib/dinstaller/dbus/manager.rb

```diff
@@ -20,77 +20,100 @@
 # find current contact information at www.suse.com.
 
 require "dbus"
-require "dinstaller/manager"
 
 module DInstaller
   module DBus
-    # YaST D-Bus object (/org/opensuse/DInstaller/Manager1)
-    #
-    # @see https://rubygems.org/gems/ruby-dbus
+    # D-Bus object to manage the installation process
     class Manager < ::DBus::Object
       PATH = "/org/opensuse/DInstaller/Manager1"
       private_constant :PATH
 
       MANAGER_INTERFACE = "org.opensuse.DInstaller.Manager1"
       private_constant :MANAGER_INTERFACE
 
-      attr_reader :logger
-
-      # @param backend [Manager] Installation manager
+      # Constructor
+      #
+      # @param backend [DInstaller::Manager] Installation manager
       # @param logger [Logger]
       def initialize(backend, logger)
         @logger = logger
         @backend = backend
-        @backend.progress.add_on_change_callback do
-          PropertiesChanged(MANAGER_INTERFACE, { "Progress" => progress }, [])
-        end
-        @backend.add_status_callback do
-          PropertiesChanged(MANAGER_INTERFACE, { "Status" => @backend.status.id }, [])
-        end
+
+        add_status_callback
+
+        add_progress_callback
 
         super(PATH)
       end
 
       dbus_interface MANAGER_INTERFACE do
-        dbus_method :Probe, "" do
-          @backend.probe
-        end
+        dbus_method(:Probe, "") { backend.probe }
 
-        dbus_method :Commit, "" do
-          @backend.install
-        end
+        dbus_method(:Commit, "" ) { backend.install }
 
-        # Enum list for statuses. Possible values:
-        # 0 : error ( it can be read from progress message )
-        # 1 : probing
-        # 2 : probed
-        # 3 : installing
-        # 4 : installed
+        # Current status
+        #
+        # Possible values:
+        #   0 : error ( it can be read from progress message )
+        #   1 : not probed
+        #   2 : probing
+        #   3 : probed
+        #   4 : installing
+        #   5 : installed
         dbus_reader :status, "u"
 
         # Progress has struct with values:
-        # s message
-        # t total major steps to do
-        # t current major step (0-based)
-        # t total minor steps. Can be zero which means no minor steps
-        # t current minor step
+        #   s message
+        #   t total major steps to do
+        #   t current major step (0-based)
+        #   t total minor steps. Can be zero which means no minor steps
+        #   t current minor step
         dbus_reader :progress, "(stttt)"
       end
 
+      # @see DInstaller::Manager#status
       def status
-        @backend.status.id
+        backend.status.id
       end
 
+      # @see DInstaller::Manager#progress
       def progress
-        prg = @backend.progress
+        progress = backend.progress
```

Comment:
well, reason I use prg is that it was too much progress for me. progress as method name, backend method and also variable names feel too confusing for me.

Reply:
Ok, I can revert it if you feel more confortable with `prg`. In general, I prefer to use complete works to make the code more readable, but it is true that threre is too many `progress` there.

Reply:
or use prefix like `backend_progress` or `progress_o` to emphasize that it is progress object.